pallets/file-system/src/utils.rs

```diff
@@ -237,13 +361,13 @@ where
             Some(mut metadata) => {
                 // Remove BSP from storage request and challenge if has confirmed having stored this file.
                 if let Some(bsp) = <StorageRequestBsps<T>>::get(&location, &who) {
-                    // TODO: challenge the BSP to force update its storage
-
                     if bsp.confirmed {
```

Comment:
We should handle the case where a BSP is NOT confirmed, but still in the storage request as volunteer. This is not an expected behaviour, as it would mean that a BSP volunteered but it is trying to remove itself as BSP, before it even provided a proof of including that file in its Merkle Patricia Forest (i.e. confirm). We can handle it with an `expect_or_err` or similar.

Reply:
We already handle this by removing the BSP from the storage request so they are no longer considered to be a volunteer.

Reply:
We will only verify the proof and update the root of the provider if they are confirmed to be storing the file of a storage request or when there is no storage request at all.